packages/ui/src/components/TaskCard/TaskCard.tsx

```diff
@@ -0,0 +1,138 @@
+import { cloneElement, ReactElement, ReactNode } from "react";
+
+import { IconType } from "../../types";
+import { LinkProps } from "../Link";
+import { HeadingLevel } from "../Heading";
+import { headingStyles } from "../Heading/styles";
+
+export const colorOptions = [
+  "primary",
+  "secondary",
+  "tertiary",
+  "quaternary",
+  "quinary",
+] as const;
+
+type CardColor = (typeof colorOptions)[number];
+
+const headingBarStyleMap: Record<CardColor, Record<string, string>> = {
+  primary: {
+    "data-h2-background-color": "base(primary.lightest)",
+    "data-h2-border-bottom": "base(primary.darkest)",
+  },
+  secondary: {
+    "data-h2-background-color": "base(secondary.lightest)",
+    "data-h2-border-bottom": "base(secondary.darkest)",
+  },
+  tertiary: {
+    "data-h2-background-color": "base(tertiary.lightest)",
+    "data-h2-border-bottom": "base(tertiary.darkest)",
+  },
+  quaternary: {
+    "data-h2-background-color": "base(quaternary.lightest)",
+    "data-h2-border-bottom": "base(quaternary.darkest)",
+  },
+  quinary: {
+    "data-h2-background-color": "base(quinary.lightest)",
+    "data-h2-border-bottom": "base(quinary.darkest)",
+  },
+};
+
+const wrapperStyleMap: Record<CardColor, Record<string, string>> = {
+  primary: {
+    "data-h2-color": "base(primary.darkest)",
+  },
+  secondary: {
+    "data-h2-color": "base(secondary.darkest)",
+  },
+  tertiary: {
+    "data-h2-color": "base(tertiary.darkest)",
+  },
+  quaternary: {
+    "data-h2-color": "base(quaternary.darkest)",
+  },
+  quinary: {
+    "data-h2-color": "base(quinary.darkest)",
+  },
+};
+
+export interface TaskCardProps {
+  icon?: IconType;
+  title: ReactNode;
+  headingColor?: CardColor;
+  link?: ReactElement<LinkProps>;
+  headingAs?: HeadingLevel;
+  children?: ReactNode;
+}
+
+const TaskCard = ({
+  icon,
+  title,
+  headingColor = "primary",
+  link: rawLink,
+  headingAs = "h3",
+  children,
+}: TaskCardProps) => {
+  // prepare link
+  const headingBarLink = rawLink
+    ? cloneElement(rawLink, {
+        color: headingColor,
+      })
+    : null;
```

Comment:
This is an [old practice](https://react.dev/reference/react/cloneElement) of React. To achieve the same result maybe change the prop to `link?: { href: string; label: string}`. Then you can use the Link component here directly, which should prevent the link from visual inconsistencies.

```tsx
<Link mode="text" href={link?.href} color={headingColor}>
    {link?.label}
 </Link>
```